### PR TITLE
Cassandra - Support pushdown for date, tinyint and smallint types

### PR DESCRIPTION
**Pull request by sundeeparun**
This is my first open source contribution, please let me know in case of any suggestions.  

Fixes https://github.com/prestosql/presto/issues/3763

### CODE REVIEW

**Comment by ebyhr** [comment]
@cla-bot check

File: presto-cassandra/src/main/java/io/prestosql/plugin/cassandra/CassandraType.java [hunk 1]
```diff
@@ -327,6 +327,7 @@ public String toCqlLiteral(Object prestoNativeValue)
             case ASCII:
             case TEXT:
             case VARCHAR:
+            case DATE:
```
Comment (findepi) [inline presto-cassandra/src/main/java/io/prestosql/plugin/cassandra/CassandraType.java:330]: Is it OK to send day number to cassandra, or should we rather send a string in yyyy-mm-dd format?

File: presto-cassandra/src/main/java/io/prestosql/plugin/cassandra/CassandraType.java [hunk 2]
```diff
@@ -434,9 +435,12 @@ public boolean isSupportedPartitionKey()
             case DOUBLE:
             case INET:
             case INT:
+            case SMALLINT:
+            case TINYINT:
             case FLOAT:
             case DECIMAL:
             case TIMESTAMP:
+            case DATE:
```
Comment (findepi) [inline presto-cassandra/src/main/java/io/prestosql/plugin/cassandra/CassandraType.java:443]: Please add tests where date, smallint and tinyint is used as partition key.

File: presto-cassandra/src/main/java/io/prestosql/plugin/cassandra/CassandraType.java [hunk 3]
```diff
@@ -299,7 +299,7 @@ public String getColumnValueForCql(Row row, int position)
             case TIMESTAMP:
                 return Long.toString(row.getTimestamp(position).getTime());
             case DATE:
-                return row.getDate(position).toString();
+                return quoteStringLiteral(row.getDate(position).toString());
```
Comment (findepi) [inline presto-cassandra/src/main/java/io/prestosql/plugin/cassandra/CassandraType.java:302]: Seems this fixes something, not adds support for, would it be possible to have a test for this?

Also, perhaps this should be a separate commit

File: presto-cassandra/src/main/java/io/prestosql/plugin/cassandra/CassandraType.java (hunk 1 above)
Comment (sundeeparun) [reply presto-cassandra/src/main/java/io/prestosql/plugin/cassandra/CassandraType.java:330]: Thanks for the review. Cassandra expects quoted String in yyyy-MM-dd format

File: presto-cassandra/src/main/java/io/prestosql/plugin/cassandra/CassandraType.java (hunk 2 above)
Comment (sundeeparun) [reply presto-cassandra/src/main/java/io/prestosql/plugin/cassandra/CassandraType.java:443]: I looked into https://github.com/prestosql/presto/pull/141 and did not find any tests added when the support for these types were added for the first time. I tested them locally by upgrading the Cassandra version in docker container as the 2.2 version does not support these types. Is it okay to upgrade the Cassandra version to 3.11 and add tests ?

File: presto-cassandra/src/main/java/io/prestosql/plugin/cassandra/CassandraType.java (hunk 3 above)
Comment (sundeeparun) [reply presto-cassandra/src/main/java/io/prestosql/plugin/cassandra/CassandraType.java:302]: This is fixing the partition key push down use case for date type and it is needed to resolve this issue

File: presto-cassandra/src/main/java/io/prestosql/plugin/cassandra/CassandraType.java (hunk 2 above)
Comment (findepi) [reply presto-cassandra/src/main/java/io/prestosql/plugin/cassandra/CassandraType.java:443]: > Is it okay to upgrade the Cassandra version to 3.11 and add tests ?

yes, i think so. @ebyhr?

Comment (ebyhr) [reply presto-cassandra/src/main/java/io/prestosql/plugin/cassandra/CassandraType.java:443]: Where are you going to add tests? The Cassandra version in the product test already supports the type as #141.

Comment (sundeeparun) [reply presto-cassandra/src/main/java/io/prestosql/plugin/cassandra/CassandraType.java:443]: Thanks for the clarification. I was referring to the test inside the Cassandra connector module. I just had a chance to look into the product tests and noticed many tests. I will add it to presto-product-tests and update this PR. Once again, Thanks a lot for your support since the beginning..

Comment (ebyhr) [reply presto-cassandra/src/main/java/io/prestosql/plugin/cassandra/CassandraType.java:443]: @sundeeparun We upgraded Cassandra docker version to 2.2 in other commit. Could you rebase and add a test again?

**Comment by noahprince22** [comment]
@sundeeparun Any updates on this? Anything I can do to help get this merged?

**Comment by adamjshook** [comment]
@sundeeparun I am hitting this issue as well and could benefit from this patch. Anything I can do to help move this along?

**Comment by ebyhr** [comment]
Superseded by #5150
